common/lib/protocol-definitions/src/clients.ts

```diff
@@ -5,53 +5,137 @@
 
 import { IUser } from "./users";
 
+/**
+ * A client's connection mode - either view-only ("read") or allowing edits ("write").
+ *
+ * @remarks Note: this should not be confused with user permissions.
+ * If a user does not have write privileges, they will still not be permitted to make edits, even if the
+ * connection mode is set to "write".
+ */
 export type ConnectionMode = "write" | "read";
 
+/**
+ * TODO
+ */
 export interface ICapabilities {
+	/**
+	 * Indicates if the client represents a potentially interactive session with a user (if 'true') or if it's a "system entity" (if 'false').
+	 *
+	 * @remarks
+	 *
+	 * The only "system entity" scenario at the moment is the
+	 * {@link https://fluidframework.com/docs/concepts/summarizer/ | summarizer client}.
+	 */
 	interactive: boolean;
 }
+
+/**
+ * TODO
+ */
 export interface IClientDetails {
+	/**
+	 * {@inheritDoc ICapabilities}
+	 */
 	capabilities: ICapabilities;
+
+	/**
+	 * TODO: What is this? Are there specific expected values? What does it mean for this to be undefined?
+	 */
 	type?: string;
+
 	/**
 	 * If the environment needs to specify multiple properties which gives info about the environment, then
-	 * it should be in particular format like: "prop1:val1;prop2:val2;prop3:val3"
+	 * it should be in particular format like: `prop1:val1;prop2:val2;prop3:val3`.
+	 *
+	 * TODO: What does it mean for this to be undefined?
 	 */
 	environment?: string;
+
+	/**
+	 * TODO: What is this? What does it mean for this to be undefined?
+	 */
 	device?: string;
 }
 
+/**
+ * TODO
+ */
 export interface IClient {
+	/**
+	 * {@inheritDoc ConnectionMode}
+	 */
 	mode: ConnectionMode;
+
+	/**
+	 * {@inheritDoc IClientDetails}
+	 */
 	details: IClientDetails;
+
+	/**
+	 * TODO: What is this? Are there specific values we expect?
+	 */
 	permission: string[];
+
+	/**
+	 * The user associated with this client.
+	 *
+	 * @remarks There may be more than 1 client associated with the same user.
+	 */
 	user: IUser;
+
+	/**
+	 * TODO: What is this? Are there specific values we expect?
+	 */
 	scopes: string[];
 
 	/**
-	 * The time the client connected
+	 * The time the client connected to the service.
+	 *
+	 * TODO: what does it mean for this to be undefined?
 	 */
 	timestamp?: number;
 }
 
+/**
+ * A client that has been acknowledged by the sequencer.
+ */
 export interface ISequencedClient {
+	/**
+	 * The underlying client details.
+	 */
 	client: IClient;
 
+	/**
+	 * The sequence number of the "join" message sent when the client joined the session.
+	 */
 	sequenceNumber: number;
 }
 
+/**
+ * TODO
+ */
 export interface ISignalClient {
+	/**
+	 * The {@link ISignalClient.client}'s unique ID.
+	 */
 	clientId: string;
 
+	/**
+	 * The underlying client details.
+	 */
 	client: IClient;
 
 	/**
-	 * Counts the number of signals sent by the client
+	 * Counts the number of signals sent by the client.
+	 *
+	 * TODO: What does it mean for this to be undefined?
```

Comment:
ODSP will always provide latest sequence number when client connected. That said, I assume this structure is used only for join & leave signals (what I've said applies only to that case).
I believe this is not implemented for FRS, but I might be wrong - worth checking.

Reply:
From what I can tell, neither ODSP nor AFR use these two properties (`clientConnectionNumber` and `referenceSequenceNumber`). They use other types with these properties, but do not set them in nor read them from `ISignalClient`s. 

`routerlicious` exposes a `ISequencedSignalClient` type with similar, required properties that appear to be used. But that's the closest thing I can find.

Maybe these are safe to deprecate? Are there any folks you know of who would be good to verify this with?